omnigibson/object_states/toggle.py

```diff
@@ -41,30 +47,58 @@ def _set_value(self, new_value):
     def _initialize(self):
         super()._initialize()
         self.initialize_link_mixin()
-        mesh_prim_path = f"{self.link.prim_path}/visual_marker"
+        mesh_prim_path = f"{self.link.prim_path}/mesh_0"
+        pre_existing_mesh = get_prim_at_path(mesh_prim_path)
         # Create a primitive mesh if it doesn't already exist
-        if not get_prim_at_path(mesh_prim_path):
+        if not pre_existing_mesh:
             mesh = create_primitive_mesh(
                 prim_path=mesh_prim_path,
                 primitive_type="Sphere",
-                extents=m.TOGGLE_BUTTON_SCALE,
+                extents=1.0,
             )
+            self.radius = m.DEFAULT_RADIUS if self.radius is None else self.radius
+        else:
+            # Infer radius from mesh if not specified as an input
+            self.radius = pre_existing_mesh.GetAttribute("radius").Get() if self.radius is None else self.radius
 
         # Create the visual geom instance referencing the generated mesh prim
         self.visual_marker = VisualGeomPrim(prim_path=mesh_prim_path, name=f"{self.obj.name}_visual_marker")
-        self.visual_marker.scale = 1 / self.obj.scale
+        self.visual_marker.scale = 2 * self.radius
         self.visual_marker.initialize()
 
         # Make sure the marker isn't translated at all
         self.visual_marker.set_local_pose(translation=np.zeros(3), orientation=np.array([0, 0, 0, 1.0]))
 
+        # Store the projection mesh's IDs
+        projection_mesh_ids = PhysicsSchemaTools.encodeSdfPath(self.visual_marker.prim_path)
+
+        # Define function for checking overlap
+        valid_hit = False
+
+        def overlap_callback(hit):
+            nonlocal valid_hit
+            valid_hit = hit.rigid_body in self._robot_link_paths
+            # Continue traversal only if we don't have a valid hit yet
+            return not valid_hit
+
+        def check_overlap():
+            nonlocal valid_hit
+            valid_hit = False
+            og.sim.psqi.overlap_shape(*projection_mesh_ids, reportFn=overlap_callback)
+            return valid_hit
+
+        self._check_overlap = check_overlap
```

Comment:
Do we not have a helper function like this in our codebase?

Reply:
nope...it's a bit tricky, because a lot of these variables can change between calls. e.g.: `self._robot_link_paths` might not always be the same each time.